src/pricer.py

```diff
@@ -371,11 +371,58 @@ def calc_rho(self, method: GreekMethod) -> float:
         raise NotImplementedError('Greeks are not implemented for tree method yet.')
 
 
-# todo: to be implemented
 class GenericPDEPricer(Pricer):
+    def __init__(self, contract: EuropeanContract, model: MarketModel, params: PDEParams):
```

Comment:
Annotations just indicate what type is expected, but they don't force it. It's just a hint in python. E.g. user will be able to pass any contract and your pricer will return a potentially incorrect price.
Have a look at EuropeanAnalyticPricer on how to type check and throw.

Reply:
Actually, if we end up with supporting Europeans only, we might want to rename this class to EuropeanPDEPricer. However, I believe that at some point it should be very easy to extend it to Forward and American.

Reply:
I would rename the class to EuropeanPDEPricer as it supports European contracts only, however both BSVol and FlatVol is supported.